### PR TITLE
Fix issues that prevented cluster tests from running concurrently.

### PR DESCRIPTION
**Pull request by nihohit**
1. tests with mocks shared the same handle, so they dropped each other's state.
2. tests with real cluster shared the same ports, so the cluster couldn't run concurrently.

### CODE REVIEW

File: redis/tests/support/cluster.rs [hunk 1]
```diff
@@ -72,6 +73,36 @@ fn port_in_use(addr: &str) -> bool {
     socket.connect(&socket_addr.into()).is_ok()
 }
 
+pub struct RedisClusterConfiguration {
+    pub nodes: u16,
```
Comment (jaymell) [inline redis/tests/support/cluster.rs:77]:
```suggestion
    pub num_nodes: u16,
```

File: redis/tests/support/cluster.rs [hunk 2]
```diff
@@ -72,6 +73,36 @@ fn port_in_use(addr: &str) -> bool {
...
+    pub replicas: u16,
```
Comment (jaymell) [inline redis/tests/support/cluster.rs:78]:
```suggestion
    pub num_replicas: u16,
```

**Comment by jaymell** [comment]
Any idea what's up with the test failure?

**Comment by nihohit** [comment]
> Any idea what's up with the test failure?

No. I don't know if it's something introduced by this change, or the old flakeyness of the cluster tests. :(

**Review by jaymell — APPROVED** [review]

**Pull request merged**